File: agent/python/mango_agent/persistence/debug.py
"""Structured debug logger — never raises."""

from __future__ import annotations

import sys
from typing import Any


def debug(scope: str, message: str, detail: Any = None) -> None:
    try:
        line = f"[mango:{scope}] {message}"
        if detail is not None:
            print(line, detail, file=sys.stderr, flush=True)
        else:
            print(line, file=sys.stderr, flush=True)
    except Exception:
        pass

File: apps/electron/src/main/lib/debug.ts
/** Structured debug logger for graceful degradation (never throws). */

export function debug(scope: string, message: string, detail?: unknown): void {
  try {
    if (detail !== undefined) {
      console.debug(`[mango:${scope}] ${message}`, detail);
    } else {
      console.debug(`[mango:${scope}] ${message}`);
    }
  } catch {
    /* ignore */
  }
}
